structuralcodes/codes/mc2010/_concrete_punching.py

```diff
@@ -0,0 +1,579 @@
+"Covers punching in Model code 2010, 7.3.5.1 to 7.3.5.4"
+import warnings
+from math import pi, sin, cos
+
+
+def b_0(v_ed: float, v_prep_d_max: float) -> float:
+    """Gives the general output for b_0, shear-resisting control perimeter.
+
+    fib Model Code 2010, eq. (7.3-57)
+
+    Args:
+        V_ed (float): The acting shear force from the columns
+        v_prep_d_max (float): The maximum shear force per unit length
+        perpendiculerer to the basic control parameter (Figure 7.3-24)
+
+    Return:
+        The shear-resisting control perimeter, b_0"""
+
+    return v_ed / v_prep_d_max
+
+
+def m_ed(
+    v_ed: float,
+    e_u: float,
+    l_x: float,
+    l_y: float,
+    l_min: float,
+    inner: bool,
+    edge_par: bool,
+    edge_per: bool,
+    corner: bool,
+) -> float:
+    """The average bending moment acting in the support strip.
+
+    fib Model Code 2010, eq. (7.3-76), (7.3-71), (7.3-72), (7.3-73)
+    and (7.3-74)
+
+    Args:
+        v_ed (float): The acting shear force from the columns
+        e_u (float): Refers to the eccentricity of the resultant of shear
+        forces with respect to the centroid
+        l_x (float): The width in x direction that the collumn carries
+        l_y (float): The width in x direction that the collumn carries
+        l_min (float): The shorter side of the the L_x and L_y
```

Comment:
If `l_min` is supposed to be `min(l_x, l_y)`, there is no need to have it as an argument, simply check for the smaller in the calculations below.

Reply:
Done